pkg/util/hostname.go

```diff
@@ -209,7 +244,7 @@ func GetHostnameData() (HostnameData, error) {
 		}
 	}
 
-	if hostName == "" {
+	if isOSHostnameUsable() && hostName == "" {
```

Comment:
Nit: if the call to `isOSHostnameUsable` either:

* has any chance to return inconsistent results between 2 calls (because of network calls/flakiness), or
* is "slow" (basically relies on network calls)

then for consistency and performance I think it'd be better to store this function's return value once at the beginning of the function